components/nif/nifstream.hpp

```diff
@@ -41,6 +41,25 @@ template <uint32_t numInstances, typename T> inline void readLittleEndianBufferO
 #endif
 }
 
+template <typename T> inline void readLittleEndianDynamicBufferOfType(Files::IStreamPtr &pIStream, T* dest, uint32_t numInstances)
+{
+#if defined(__x86_64__) || defined(_M_X64) || defined(__i386) || defined(_M_IX86)
+    pIStream->read((char*)dest, numInstances * sizeof(T));
+#else
+    char buffer[numInstances * sizeof(T)];
+    pIStream->read((char*)buffer, numInstances * sizeof(T));
+    /*
+    Due to the loop iterations being known at compile time,
+    this nested loop will most likely be unrolled
+    */
+    for (uint32_t i = 0; i < numInstances; i++)
+    {
+        dest[i] = 0;
+        for (uint32_t byte = 0; byte < sizeof(T); byte++)
+            dest[i] |= ((T)buffer[i * sizeof(T) + byte]) << (byte * 8);
```

Comment:
Does this actually work for the ```osg::Vec*``` types?
```
osg::Vec3f *dest;
...
dest[i] |= ((osg::Vec3f)buffer[i * sizeof(osg::Vec3f) + byte]) << (byte * 8);
```
looks extremely sketchy.

Reply:
This template should be used sparingly.  It only works for guaranteed packed objects.  The developer of OSG did a good job defining these objects.  If you look at the definition, the data for the individual floats is stored as an array.  It's guaranteed to be packed.  If we start deriving from osg::Vec3f, the object may have more storage and then this would break but I don't see examples of that in openmw that i'm aware of.  We could put static assertions that the osg vec types equal the size of the nif counterpart for them.